crates/biome_js_analyze/src/lint/nursery/use_jsx_curly_brace_convention.rs

```diff
@@ -0,0 +1,348 @@
+                        } else {
+                            None
+                        }
+                    })?;
+                    let text = &str_literal.value_token().ok()?.token_text();
+                    // trim the quotes off of the string literal
+                    let text_trimmed = text.clone().slice(TextRange::new(
+                        1.into(),
+                        text.len().checked_sub(1.into()).unwrap_or(text.len()),
+                    ));
+                    let jsx_text = biome_js_syntax::AnyJsxChild::JsxText(make::jsx_text(
+                        JsSyntaxToken::new_detached(
+                            JsSyntaxKind::JS_STRING_LITERAL,
+                            &format!("{text_trimmed}"),
+                            [],
+                            [],
+                        ),
+                    ))
+                    .into_syntax()
+                    .into();
+                    mutation.replace_element(node.clone().into_syntax().into(), jsx_text);
+                }
+            }
+        }
+
+        let msg = match state {
+            CurlyBraceResolution::AddBraces => "Add curly braces around the expression.",
+            CurlyBraceResolution::RemoveBraces => "Remove curly braces around the expression.",
+        };
+
+        Some(JsRuleAction::new(
+            ActionCategory::QuickFix,
+            ctx.metadata().applicability(),
+            markup! { {msg} }.to_owned(),
+            mutation,
+        ))
+    }
+}
+
+fn handle_attr_init_clause(
+    attr: &JsxAttributeInitializerClause,
+    has_curly_braces: bool,
+) -> Option<CurlyBraceResolution> {
+    let Ok(node) = attr.value() else {
+        return None;
+    };
+
+    match node {
+        AnyJsxAttributeValue::AnyJsxTag(_) => Some(CurlyBraceResolution::AddBraces),
+        AnyJsxAttributeValue::JsxExpressionAttributeValue(node) => {
+            if has_curly_braces && contains_string_literal(&node) {
+                Some(CurlyBraceResolution::RemoveBraces)
+            } else if !has_curly_braces && contains_jsx_tag(&node) {
+                Some(CurlyBraceResolution::AddBraces)
+            } else {
+                None
+            }
+        }
+        AnyJsxAttributeValue::JsxString(_) => None,
+    }
+}
+
+fn handle_jsx_child(child: &AnyJsxChild, has_curly_braces: bool) -> Option<CurlyBraceResolution> {
+    match child {
+        AnyJsxChild::JsxExpressionChild(child) => child
+            .expression()
+            .as_ref()
+            .and_then(|node| node.as_any_js_literal_expression())
+            .and_then(|node| node.as_js_string_literal_expression())
+            .and({
+                if has_curly_braces {
+                    Some(CurlyBraceResolution::RemoveBraces)
+                } else {
+                    None
+                }
+            }),
+        AnyJsxChild::JsxText(_) => None,
+        _ => None,
+    }
+}
+
+fn has_curly_braces(node: &UseJsxCurlyBraceConventionQuery) -> bool {
```

Comment:
perhaps this version might be more readable? How do you think:) @dyc3 

```
fn has_curly_braces(node: &UseJsxCurlyBraceConventionQuery) -> bool {
    match node {
        UseJsxCurlyBraceConventionQuery::JsxAttributeInitializerClause(node) => {
            node.value()
                .map(|node| matches!(node, AnyJsxAttributeValue::JsxExpressionAttributeValue(attr) if attr.l_curly_token().is_ok() || attr.r_curly_token().is_ok()))
                .unwrap_or(false)
        }
        UseJsxCurlyBraceConventionQuery::AnyJsxChild(node) => match node {
            AnyJsxChild::JsxExpressionChild(node) => node.l_curly_token().is_ok() || node.r_curly_token().is_ok(),
            AnyJsxChild::JsxSpreadChild(_) => true,
            _ => false,
        },
    }
}
```

Reply:
Yeah that looks a little better.